### DISCUSSION TITLE
Your Guide to Soroban-RPC Requests

### DISCUSSION THREAD

**Discussion by ElliotFriend** [stellar/sorobanathon#113] [General]
Under the hood, Soroban-RPC is a JON-RPC server that is listening for (and then responding) to specific requests. While `soroban-cli` can abstract a lot of the confusion away from making these requests, it's still helpful to have an understanding of what is going on beneath the surface.

With that in mind, I've created a Soroban-RPC guide outlining which methods are available, what parameters those methods expect, how to create/encode those parameters, etc.

I've also included an import-able Postman collection that gets those requests set up for anybody who's partial to using that. Additionally, there are a couple starter scripts to help folks understand how to make these kinds of requests using python/javascript SDKs.

> This is still very much an evolving guide. Some methods are incompletely documented, and others I've not been able to test run yet. I've tried to make everything as up-to-date as I can, but will continue to keep this current as soroban development progresses.

Here's the repository: https://github.com/ElliotFriend/soroban-rpc-postman

Here's the Postman collection:

[![Run in Postman](https://run.pstmn.io/button.svg)](https://app.getpostman.com/run-collection/5352332-9ccf291a-944a-4037-8655-1d6c35d20cf8?action=collection%2Ffork&collection-url=entityId%3D5352332-9ccf291a-944a-4037-8655-1d6c35d20cf8%26entityType%3Dcollection%26workspaceId%3D0450a14f-82ac-4faf-8045-b7bcf89928b5)

I'd love to hear any feedback! Thanks!!

**Comment by achallagundla** [comment]
Heya,
Congratulations! Your contribution to Sorobanathon: First Light has been accepted! 🎉
Your submission may qualify for a reward of 1100 XLM! One last step: to receive that reward, please fill out this [form](https://stellarform.typeform.com/Sorobanathon) completely and accurately.

Please let us know if you have any questions. (You can also message me on discord @ anuxhya)

All the best,
Anuhya

PS: No need to submit the form if you already have, just update me on discord if your wallet address has changed.
